cmd/thanos/receive.go

```diff
@@ -461,10 +465,12 @@ func setupHashring(g *run.Group,
 		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(func() error {
 			level.Info(logger).Log("msg", "the hashring initialized with config watcher.")
-			return receive.HashringFromConfigWatcher(ctx, algorithm, conf.replicationFactor, updates, cw)
+			return nil, receive.HashringFromConfigWatcher(ctx, algorithm, conf.replicationFactor, updates, cw)
```

Comment:
This is within anonymous func passed to `g.Add` so it has a function signature of `func() error`. So only errors need to be returned.
```suggestion
			return receive.HashringFromConfigWatcher(ctx, algorithm, conf.replicationFactor, updates, cw)
```